src/olympia/promoted/models.py

```diff
@@ -127,3 +129,47 @@ def update_es_for_promoted(sender, instance, **kw):
 def update_es_for_promoted_approval(sender, instance, **kw):
     update_es_for_promoted(
         sender=sender, instance=instance.version, **kw)
+
+
+class PromotedSubscription(ModelBase):
+    promoted_addon = models.OneToOneField(
+        PromotedAddon,
+        on_delete=models.CASCADE,
+        null=False,
+    )
+    link_visited_at = models.DateTimeField(
+        null=True,
+        help_text=(
+            "This date is set when the developer has visited the onboarding "
+            "page.",
+        ),
+    )
+    stripe_session_id = models.CharField(
+        default=None, null=True, max_length=100
+    )
+    payment_cancelled_at = models.DateTimeField(
```

Comment:
Both `DateTimeField`s are probably a bit too simplistic in the long run - we'll have to define exactly what happens when developers cancel and come back, for instance but I reckon this is enough to get us started. (And we might not need to store too much ourselves, ultimately Stripe does have that data available through their API)

Reply:
Yes, I agree.

According to the PRD, we want to offer the developer to pay again (and again and again..) when they cancel on Stripe Checkout (which is only possible with the "back" arrow/button). It's a bit different than cancelling the subscription after the first payment, but this is not covered by the PRD.